### ISSUE TITLE
Predict uses test set, evaluate uses validation set

### ISSUE THREAD

**Issue by kl2792** [ecs-vlc/torchbearer#374]
I might be misunderstanding something about the Trial API design -- but do you think it should be possible to predict on the validation set or evaluate on the test set (assuming you have labels)? 

What are your thoughts on having an optional keyword to predict/evaluate on an arbitrary DataLoader (aside from the loaded train/dev/test sets)?

Thanks!

**Comment by ethanwharris** [comment]
Would be nice to take the state_key which points to the generator as an optional argument for predict and evaluate. Would need to alias the generator keys so that the API is cleaner. E.g. (in state.py)
`TRAIN_DATA = TRAIN_GENERATOR = state_key(...)`
